homeassistant/components/izone/climate.py

```diff
@@ -0,0 +1,468 @@
+            self.hass, self.supply_temperature, self.temperature_unit,
+            self.precision)
+        data['temp_setpoint'] = show_temp(
+            self.hass, self._controller.temp_setpoint,
+            self.temperature_unit, self.precision)
+        return data
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        if not self._controller.is_on:
+            return STATE_OFF
+        mode = self._controller.mode
+        for (key, value) in self._state_to_pizone.items():
+            if value == mode:
+                return key
+        assert False, "Should be unreachable"
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [STATE_OFF] + list(self._state_to_pizone.keys())
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if not self._available:
+            return None
+        if self._controller.mode == Controller.Mode.FREE_AIR:
+            return self._controller.temp_supply
+        return self._controller.temp_return
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if not self._supported_features & SUPPORT_TARGET_TEMPERATURE:
+            return None
+        return self._controller.temp_setpoint
+
+    @property
+    def supply_temperature(self):
+        """Return the current supply, or in duct, temperature."""
+        return self._controller.temp_supply
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 0.5
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._controller.is_on
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._controller.fan.name.title()
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return list(self._fan_to_pizone.keys())
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._controller.temp_min
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._controller.temp_max
+
+    async def _wrap_and_catch(self, coro):
+        try:
+            await coro
+            self.set_available(True)
+        except ConnectionError:
+            self.set_available(False)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if not self.supported_features & SUPPORT_TARGET_TEMPERATURE:
+            self.async_schedule_update_ha_state(True)
+            return
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp:
+            return self._wrap_and_catch(
```

Comment:
```suggestion
            await self._wrap_and_catch(
```

I don't think it will work with a return - otherwise the coroutine will never be executed I think

Reply:
You're right... but it's actually working which is odd.